frame/balances/src/lib.rs

```diff
@@ -610,16 +610,17 @@ impl<T: Config<I>, I: Instance> Module<T, I> {
 	fn post_mutation(
 		who: &T::AccountId,
 		new: AccountData<T::Balance>,
-	) -> Option<AccountData<T::Balance>> {
+	) -> (Option<AccountData<T::Balance>>, Option<NegativeImbalance<T, I>>) {
 		let total = new.total();
 		if total < T::ExistentialDeposit::get() {
-			if !total.is_zero() {
-				T::DustRemoval::on_unbalanced(NegativeImbalance::new(total));
+			if total.is_zero() {
+				(None, None)
+			} else {
 				Self::deposit_event(RawEvent::DustLost(who.clone(), total));
```

Comment:
should we move emit DustLost event to after `DustRemoval::on_unbalanced`? previously event is emitted after handler, this make it before handler.

Reply:
Events are going to change order with the proper fix.